### DISCUSSION TITLE
mcap file is unindexed

### DISCUSSION THREAD

**Discussion by jlack1987** [foxglove/community#1053] [Q&A] [answered]
I have an mcap file containing some log data, but when I try to open it to view the data in foxglove studio it says "This file is unindexed". I haven't been able to find anywhere that says what this means or what I can do to fix the file.

**Comment by defunctzombie** [comment]
MCAP files can be written with or without an index (https://mcap.dev/spec). Files with an index work much better. How did you produce your mcap file?

**Reply by jlack1987** [reply]
I have a data logging application that's using the `mcap::McapWriter` c++ object provided by the [mcap library](https://github.com/foxglove/mcap/tree/main/cpp/mcap/include/mcap) I'm filling out the message with serialized json data [here](https://gitlab.com/jlack/data_tamer_tools/-/blob/main/src/foxglove_mcap_sink.cpp?ref_type=heads#L80).

**Reply by jlack1987** [reply]
Do I just need to increment that sequence field with each message? I'm looking at the example [here](https://github.com/foxglove/mcap/blob/main/cpp/examples/jsonschema/writer.cpp)

**Reply by defunctzombie** [reply]
When you create a new `mcap::McapWriter` you specify options (https://mcap.dev/docs/cpp/r832FE362A16BB6E8). These let you control various features of the file you write. By default _indexing_ is turned on, so your code had to explicitly turn it off.

**Reply by jlack1987** [reply]
hmm interesting. I'm creating a default constructed options struct and setting `compression` and `profile` but leaving everything else as the default 🤔

**Reply by achim-k** [reply, marked as answer]
Are you correctly closing the writer by calling [close](https://mcap.dev/docs/cpp/rF2221FA24F2318CA#62D931C0BB908B8A)() ?

**Reply by jlack1987** [reply]
It is being called in the destructor of the class that's writing the data. I should probably verify the destructor is getting called

**Reply by jlack1987** [reply]
using a python script I iterate through the messages and each line has something like,
`MessageIndex(channel_id=1, records=[(<rly_big_number>, 0)]`

**Reply by jlack1987** [reply]
It appears that that `rly_big_number` is too big. I'm not sure what the number means, but it has some relation to size of something i'm guessing bc it looks like there's a 4GB limit and mine is much bigger than that, although my log file size is only 1MB. How is that rly big number calculated?

**Reply by jlack1987** [reply]
I finally found the problem here and you were right about not calling `close()`. I am calling `close` on destruction, but the code wasn't being killed properly. Once I fixed that this problem went away
